ldm/modules/kohya_lora_manager.py

```diff
@@ -0,0 +1,367 @@
+                    self.rank = value.shape[0]
+                self.load_lora_layer(stem, leaf, value, wrapped)
+                continue
+            else:
+                continue
+
+    def load_lora_layer(self, stem: str, leaf: str, value, wrapped: torch.nn.Module):
+        layer = self.layers.get(stem, None)
+        if layer is None:
+            layer = LoRALayer(self.name, stem, self.rank, self.alpha)
+            self.layers[stem] = layer
+
+        if type(wrapped) == torch.nn.Linear:
+            module = torch.nn.Linear(value.shape[1], value.shape[0], bias=False)
+        elif type(wrapped) == torch.nn.Conv2d:
+            module = torch.nn.Conv2d(value.shape[1], value.shape[0], (1, 1), bias=False)
+        else:
+            print(f">> Encountered unknown lora layer module in {self.name}: {type(value).__name__}")
+            return
+
+        with torch.no_grad():
+            module.weight.copy_(value)
+
+        module.to(device=self.device, dtype=self.dtype)
+
+        if leaf == "lora_up.weight":
+            layer.up = module
+        elif leaf == "lora_down.weight":
+            layer.down = module
+        else:
+            print(f">> Encountered unknown layer in lora {self.name}: {leaf}")
+            return
+
+
+class KohyaLoraManager:
+    def __init__(self, pipe, lora_path):
+        self.unet = pipe.unet
+        self.lora_path = lora_path
+        self.wrapper = LoRAModuleWrapper(pipe.unet, pipe.text_encoder)
+        self.text_encoder = pipe.text_encoder
+        self.device = torch.device(choose_torch_device())
+        self.dtype = pipe.unet.dtype
+        self.loras_to_load = {}
+
+    def load_lora_module(self, name, path_file, multiplier: float = 1.0):
+        # can be used instead to load through diffusers, once enough support is added
+        # lora = load_lora_attn(name, path_file, self.wrapper, multiplier)
+
+        print(f"   | Found lora {name} at {path_file}")
+        if path_file.suffix == '.safetensors':
+            checkpoint = load_file(path_file.absolute().as_posix(), device='cpu')
+        else:
+            checkpoint = torch.load(path_file, map_location='cpu')
+
+        lora = LoRA(name, self.device, self.dtype, self.wrapper, multiplier)
+        lora.load_from_dict(checkpoint)
+        self.wrapper.loaded_loras[name] = lora
+
+        return lora
+
+    def configure_prompt(self, prompt: str) -> str:
+        self.clear_loras()
+
+        lora_match = re.compile(r"withLora\(([a-zA-Z_-]+),?\s*([\d.]+)?\)")
+
+        for match in lora_match.findall(prompt):
+            name, mult = match
+            name = name.strip()
+            mult = float(mult or '1.0')
+            self.set_lora(name, mult)
+
+        # remove lora and return prompt to avoid the lora prompt causing issues in inference
+        return re.sub(lora_match, "", prompt)
+
+    def apply_lora_model(self, name, mult: float = 1.0):
+        for suffix in ['ckpt','safetensors']:
```

Comment:
Same as above - maybe scan for `pt` extension too?

Reply:
Fixed. Thanks for the suggestion.